plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java

```diff
@@ -878,6 +878,8 @@ static void registerOrgQbiccRuntimeCNativeIntrinsics(final CompilationContext ct
             if (value instanceof Load) {
                 Load load = (Load) value;
                 return builder.addressOf(load.getValueHandle());
+            } else if (value.getType() instanceof PointerType) {
+                return value;
```

Comment:
I was thinking since `deref` is now just returning instance instead of loading the value that `addr_of` should do something similar since they are dereference & reference. That way `addr_of(ptr).deref()` for example returns the expected outcome. What am I missing?

Reply:
I'm out for the weekend but maybe it would be a good idea to have a call next week to figure it out, wdyt?

Reply:
Sounds great. I'll set something up.